pkg/processor/runtime/rpc/abstract.go

```diff
@@ -301,34 +244,20 @@ func (r *AbstractRuntime) Terminate() error {
 	return nil
 }
 
-func (r *AbstractRuntime) processItemAndWaitForResult(item interface{}, functionLogger logger.Logger) (*batchedResults, error) {
+func (r *AbstractRuntime) allocateSocketAndWaitForResult(item interface{}, functionLogger logger.Logger) (*batchedResults, error) {
```

Comment:
I would leave the name `processItemAndWaitForResult`, as the socket allocation is an implementation detail for processing the event.
The processing is what is actually being done here